packages/cosmic-swingset/lib/ag-solo/vats/lib-wallet.js

```diff
@@ -79,6 +83,30 @@ export async function makeWallet(
     inboxStateChangeHandler(getInboxState());
   }
 
+  async function updateOffersState(id, offerHandle, update) {
+    const { value, updateHandle, done } = update;
+    if (done) {
+      idToOfferHandle.delete(id);
+      idToOffer.delete(id);
+      offersState.delete(id);
+    } else {
+      E.when(updateHandle).then(nextUpdate =>
+        updateOffersState(id, offerHandle, nextUpdate),
+      );
+    }
+    offersState.set(id, value);
+    offersStateChangeHandler(getOffersState());
+  }
+
+  // There's a new offer. Ask Zoe to notify us when the offer is done.
+  async function subscribeToNotifier(id, offerHandle) {
+    const offerNotifier = await E(zoe).getOfferNotifier(offerHandle);
+    const { updateHandle } = offerNotifier.getUpdateSince();
+    E.when(updateHandle).then(update =>
+      updateOffersState(id, offerHandle, update),
```

Comment:
I think we want to hook this up to the inboxState, which is badly named, but that already holds the offer data that is given to the UIs. What we need to do is figure out what part of the offerRecord should be sent to the UI as well  (by updating the offers in inboxState). For example, it might just be 'active': true or 'allocation': extents